docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,159 @@
+
+### 2. AbstractParquetFileReader
+Reads Parquet files from input directory using GroupReadSupport. Derived classes need to implement [convertGroup(Group)](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/parquet/AbstractParquetFileReader.html#convertGroup(Group)) method to convert Group to other type. Also it should implement  [readEntity()](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html#readEntity()) and [emit(T)](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html#emit(T)) methods.
+
+#### Properties and Ports
+#### <a name="AbstractParquetFileReaderProps"></a>Properties of AbstractParquetFileReader
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *parquetSchema*| Parquet Schema to parse record. | String | Yes | N/A |
+
+#### Ports
+This operator has no ports.
+
+### 3. AbstractThroughputFileInputOperator
+This operator provides dynamic partitioning to AbstractFileInputOperator. The user can set the preferred number of pending files per operator as well as the max number of operators and define a repartition interval. If a physical operator runs out of files to process and an amount of time greater than or equal to the repartition interval has passed then a new number of operators are created to accommodate the remaining pending files. Derived classes need to implement [readEntity()](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html#readEntity()) and [emit(T)](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html#emit(T)) methods.
+
+#### Properties and Ports
+#### <a name="AbstractThroughputFileInputOperatorProps"></a>Properties of AbstractThroughputFileInputOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *repartitionInterval*| The minimum amount of time that must pass in milliseconds before the operator can be repartitioned. | long | No | 5 minutes |
+| *preferredMaxPendingFilesPerOperator* | the preferred number of pending files per operator. | int | No | 10 |
+| *partitionCount* | the maximum number of partitions for the operator. | int | No | 1 |
+
+#### Ports
+This operator has no ports.
+
+### 4. LineByLineFileInputOperator
+The operator read contents of a file line by line. Each line is emitted as a separate tuple in string format.
+
+#### Properties and Ports
+#### Prooperties
+This operator has no added properties. Refer [parent class](#AbstractFileInputOperatorProps) properties for more details.
+
+#### Ports
+| **Port** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *output* | Tuples that are read from file are emitted on this port | String | No |
```

Comment:
Connecting this port is mandatory; if not connected the app fails with: An error occurred trying to launch the application. Server message: javax.validation.ValidationException: At least one output port must be connected: